primitives/signed.fil

```diff
@@ -1,12 +1,30 @@
+import "comb.fil";
+
 extern "signed.sv" {
   comp NegConst[WIDTH, VALUE]<'G: 'L-('G), ?'L: 1='G+1>() -> (
      out: ['G, 'L] WIDTH,
-  ) where 'L > 'G;
+  ) where
+   'L > 'G,
+   WIDTH > 0;
 
   comp SAdd[WIDTH]<'G: 'L-('G), ?'L: 1='G+1>(
      left: ['G, 'L] WIDTH,
      right: ['G, 'L] WIDTH
   ) -> (
      out: ['G, 'L] WIDTH,
   ) where 'L > 'G;
+}
+
+comp SignedConst[WIDTH, VALUE]<'G: 1>() -> (
```

Comment:
What's the difference between signed const and neg const?

Reply:
Signed const allows us to take an unsigned `N` bit parameter value and convert it to the signed wire equivalent. `NegConst` just takes `x` and returns `-x` I think.

Reply:
One seems more expressive than the other in that case. We should remove them accordingly and pick one.